### DISCUSSION TITLE
PR for custom ID

### DISCUSSION THREAD

**Discussion by Its-Just-Nans** [node-saml/node-saml#156] [Ideas]
Hi,

Can I make a PR for a custom `ID` in the metadata ?

[https://docs.oasis-open.org/security/saml/v2.0/saml-metadata-2.0-os.pdf](https://docs.oasis-open.org/security/saml/v2.0/saml-metadata-2.0-os.pdf) line 374

In other words: change this line https://github.com/node-saml/node-saml/blob/d03d9ae519a72e7ed455bdd2a7740e61c444be86/src/metadata.ts#L54

**Comment by cjbarth** [comment]
I have no objection, but we should preserve the current functionality, which is a sensible default.

**Reply by Its-Just-Nans** [reply]
Great !
PR here : https://github.com/node-saml/node-saml/pull/157
